### PR TITLE
On-Behalf-Of Token Exchange

### PR DESCRIPTION
**Pull request by susanharper-okta**
## Description:
- **What's changed?** Adding content for On-Behalf-Of Token Exchange - 
- **Is this PR related to a Monolith release?** yes 2023.02.0

Netlify link: https://63ddd67b48ae0742e99ce475--reverent-murdock-829d24.netlify.app/docs/guides/set-up-token-exchange/main/

### Resolves:

* [OKTA-568964](https://oktainc.atlassian.net/browse/OKTA-568964)

### CODE REVIEW

**Comment by franklu-okta** [comment]
Do we have screenshot here?

**Review by franklu-okta — COMMENTED** [review]
Do we have screenshot here?

**Comment by susanharper-okta** [comment]
> Do we have screenshot here?

I'm waiting on the Netlify link and then I'll post that in the PR Description

File: packages/@okta/vuepress-site/docs/guides/set-up-token-exchange/main/index.md [hunk 1]
```diff
@@ -0,0 +1,345 @@
...
+
+Create a service app that represents API1.
+
+1. On the **Applications** page, click **Create App Integration**.
+1. Select **API Services**, and then click **Next**.
+1. Name your application. For this example, enter **API1**, and then click **Save**.
+1. In the **General Settings** section of the **General** tab, click **Edit** and select **Token Exchange** as the grant type for your application.
+1. Click **Save**.
+1. Copy the client ID in the **Client Credentials** section, and then copy the client secret in the **CLIENT SECRETS** section.
+1. [Base64 encode](/docs/guides/implement-grant-type/clientcreds/main/#flow-specifics) the client ID and client secret for use in the token exchange request from API1 to API2.
```
Comment (franklu-okta) [inline packages/@okta/vuepress-site/docs/guides/set-up-token-exchange/main/index.md:77]: The link maybe change to: 
-> `https://63ddd67b48ae0742e99ce475--reverent-murdock-829d24.netlify.app/docs/guides/implement-grant-type/clientcreds/main/#base64-encode-the-client-id-and-client-secret`

File: packages/@okta/vuepress-site/docs/guides/set-up-token-exchange/main/index.md [hunk 2]
```diff
@@ -0,0 +1,345 @@
...
+
+Add some custom scopes to both custom authorization servers to request during the token exchange calls.
+
+> **Note:** You can't include the `offline_access` scope or any of the OpenID Connect scopes when using a service app. This means that you can't request refresh tokens or ID tokens in a service app-initiated token exchange flow.
+
+1. From the Admin Console, select **Security** > **API**, and then select one of the authorization servers that you want to use.
+1. Select the **Scopes** tab, and then click **Add Scope**.
+1. In the dialog that appears, enter a scope name (**api:access:read**).
+1. Click Create.
+1. Repeat steps 1-4 and create a second custom scope called **api:access:write**.
```
Comment (franklu-okta) [inline packages/@okta/vuepress-site/docs/guides/set-up-token-exchange/main/index.md:93]: Do we need parentheses here?
-> `(**api:access:write**)`

File: packages/@okta/vuepress-site/docs/guides/set-up-token-exchange/main/index.md [hunk 3]
```diff
@@ -0,0 +1,345 @@
...
+1. Click **Create Policy**.
+1. Click **Add Rule** and in the dialog that appears, enter the following:
+
+   * **Name:** Enter a name for the rule (**Mobile app to API1**).
+   * **AND Scopes requested:** Select **The following scopes** and enter **openid**.
+
+1. Click **Create rule**.
+1. Repeat steps 1-5 to create a policy and a rule that allows the service app that represents API1 to talk to API2. Use the following values for the policy:
+
+   * **Name:** Access API2.
```
Comment (franklu-okta) [inline packages/@okta/vuepress-site/docs/guides/set-up-token-exchange/main/index.md:126]: Do we want to bold it along with the parentheses? 
-> (**Access API1**)

File: packages/@okta/vuepress-site/docs/guides/set-up-token-exchange/main/index.md [hunk 4]
```diff
@@ -0,0 +1,345 @@
...
+   * **Assign to:** Select The following clients, start typing **API1** (the service app that you created earlier) and select it from the list that appears.
+
+   Use the following values for the rule:
+   * **Name:** For this example, enter: API1 to API2.
```
Comment (franklu-okta) [inline packages/@okta/vuepress-site/docs/guides/set-up-token-exchange/main/index.md:130]: Do we want to bold it along with the parentheses?
-> (**API1 to API2**)

File: packages/@okta/vuepress-site/docs/guides/set-up-token-exchange/main/index.md [hunk 5]
```diff
@@ -0,0 +1,345 @@
...
+
+### Flow specifics with a trusted server
+
+Perform the requests in the previous [Flow specifics](#flow-specifics) section. When you reach the [token exchange request from the service app](#token-exchange-request-from-service-app-to-api) request, it should look like the following:
+
+> **Note:** Use the Base64 encoded client ID and secret for the service app that you generated during the service app creation earlier. Place that in the Authorization header. See the following example for the format.
+
+```curl
+curl --location --request POST \
+  --url 'https://${yourOktaDomain}/oauth2/{trustedauthServerId}/v1/token' \
```
Comment (franklu-okta) [inline packages/@okta/vuepress-site/docs/guides/set-up-token-exchange/main/index.md:291]: Maybe better to change to: 
-> `--url 'https://${yourOktaDomain}/oauth2/{trustedAuthServerId}/v1/token' \`

Uppercase for A in trusted**A**uthServerId.

File: packages/@okta/vuepress-site/docs/guides/set-up-token-exchange/main/index.md [hunk 6]
```diff
@@ -0,0 +1,345 @@
...
+  --data-urlencode 'subject_token={access token from the Authorization Code with PKCE flow}' \
+  --data-urlencode 'scope=api:access:read api:access:write' \
+  --data-urlencode 'audience=com.api.atko' \
+```
+
+This request is sent to the authorization server with the trusted server that you configured in the [Add a trusted server](#add-a-trusted-server) section.
+
+Properties sent in the request body:
+
+* `grant-type`: urn:ietf:params:oauth:grant-type:token-exchange
```
Comment (franklu-okta) [inline packages/@okta/vuepress-site/docs/guides/set-up-token-exchange/main/index.md:306]: -> `grant_type`

File: packages/@okta/vuepress-site/docs/guides/set-up-token-exchange/main/index.md [hunk 7]
```diff
@@ -0,0 +1,345 @@
...
+### Authorization Code with PKCE request
+
+Use the Authorization Code with PKCE flow to obtain an authorization code for the client. In this case, the mobile app requests tokens so that it can talk to API1.
+
+```cURL
+curl --location --request POST \
+  --url 'https://${yourOktaDomain}/oauth2/{authServerId}/v1/authorize' \
+  --header 'Accept: application/json' \
+  --header 'Content-Type: application/x-www-form-urlencoded' \
+  --data-urlencode 'client_id=${nativeAppclientId}' \
```
Comment (franklu-okta) [inline packages/@okta/vuepress-site/docs/guides/set-up-token-exchange/main/index.md:144]: Maybe change to: 
-> nativeAppClientId ?

File: packages/@okta/vuepress-site/docs/guides/set-up-token-exchange/main/index.md [hunk 8]
```diff
@@ -0,0 +1,345 @@
...
+```curl
+curl --location --request POST \
+  --url 'https://${yourOktaDomain}/oauth2/{authServerId}/v1/token' \
+  --header 'Accept: application/json' \
+  --header 'Content-Type: application/x-www-form-urlencoded' \
+  --data-urlencode 'grant_type=authorization_code' \
+  --data-urlencode 'redirect_uri=${configuredRedirectUri}' \
+  --data-urlencode 'code=FQGFlDO-J1j.....QvabuZ7-cfYJ0KtKB8' \
+  --data-urlencode 'code_verifier=xO5wgOEH5UA2XUdVQ88pM.....Rtc5ERKq1MeonMo8QLCSRYlDk' \
+  --data-urlencode 'client_id=${nativeAppclientId}'
```
Comment (franklu-okta) [inline packages/@okta/vuepress-site/docs/guides/set-up-token-exchange/main/index.md:170]: Maybe change to: 
-> `nativeAppClientId`?

File: packages/@okta/vuepress-site/docs/guides/set-up-token-exchange/main/index.md [hunk 9]
```diff
@@ -0,0 +1,345 @@
...
+1. Repeat steps 1-5 for the second authorization server that you want to use.
+
+#### Update audience
+
+Just for this example, update the audience of the authorization servers to help with understanding the token exchange flow when you view the responses.
+
+1. On the **API** page, select the edit (pencil) icon to the right of one of the authorization servers.
+1. In the **Settings** section, click **Edit**.
+1. Change the **Audience** to **com.api.atko** and click **Save**.
+1. Repeat steps 1-3 for the second authorization server.
```
Comment (franklu-okta) [inline packages/@okta/vuepress-site/docs/guides/set-up-token-exchange/main/index.md:103]: I am thinking, if this step will create the confusion that the audience of AS1 has to be the same as the audience of AS2. 
Maybe we can have different audiences for AS1 and AS2 to eliminate this confusion.

File: packages/@okta/vuepress-site/docs/guides/set-up-token-exchange/main/index.md (hunk 1 above)
Comment (franklu-okta) [reply packages/@okta/vuepress-site/docs/guides/set-up-token-exchange/main/index.md:77]: I mean this part: `clientcreds/main/#base64-encode-the-client-id-and-client-secret`

**Review by franklu-okta — APPROVED** [review]
LGTM.

File: packages/@okta/vuepress-site/docs/guides/set-up-token-exchange/main/index.md [hunk 10]
```diff
@@ -0,0 +1,345 @@
...
+1. Name your application and at the bottom of the page select **Allow everyone in your organization to access**.
+1. Click **Save** and at the top of the page click **Back to Applications**.
+
+> **Note:** We recommend that native applications use the Authorization Code with PKCE authentication flow. See [Implement authorization by grant type](/docs/guides/implement-grant-type/authcodepkce/main/#create-the-proof-key-for-code-exchange) for more information on creating the Proof Key for Code Exchange (PKCE) for your native application.
+
+### Create a service app
+
+In token exchange use cases, an API microservice can act both as a resource server and a client. For this example, the native app gets an access token to make API requests to API1. API1 acts as the resource server. When API1 needs to talk to API2 on behalf of the user, the API1 becomes the OAuth client.
+
+Create a service app that represents API1.
```
Comment (isaacdurham-okta) [inline packages/@okta/vuepress-site/docs/guides/set-up-token-exchange/main/index.md:69]: Is this supposed to be a heading, or something else?

File: packages/@okta/vuepress-site/docs/guides/set-up-token-exchange/main/index.md [hunk 11]
```diff
@@ -0,0 +1,345 @@
...
+Add some custom scopes to both custom authorization servers to request during the token exchange calls.
+
+> **Note:** You can't include the `offline_access` scope or any of the OpenID Connect scopes when using a service app. This means that you can't request refresh tokens or ID tokens in a service app-initiated token exchange flow.
+
+1. From the Admin Console, select **Security** > **API**, and then select one of the authorization servers that you want to use.
+1. Select the **Scopes** tab, and then click **Add Scope**.
+1. In the dialog that appears, enter a scope name (**api:access:read**).
+1. Click Create.
+1. Repeat steps 1-4 and create a second custom scope called **api:access:write**.
+1. Repeat steps 1-5 for the second authorization server that you want to use.
```
Comment (isaacdurham-okta) [inline packages/@okta/vuepress-site/docs/guides/set-up-token-exchange/main/index.md:94]:
```suggestion
1. Repeat steps 1-5 for the second authorization server.
```

File: packages/@okta/vuepress-site/docs/guides/set-up-token-exchange/main/index.md [hunk 12]
```diff
@@ -0,0 +1,345 @@
...
+
+#### Update audience
+
+Just for this example, update the audience of the authorization servers to help with understanding the token exchange flow when you view the responses.
```
Comment (isaacdurham-okta) [inline packages/@okta/vuepress-site/docs/guides/set-up-token-exchange/main/index.md:98]:
```suggestion
Just for this example, update the audience of the authorization servers to help better understand the token exchange flow when you view the responses.
```

File: packages/@okta/vuepress-site/docs/guides/set-up-token-exchange/main/index.md [hunk 13]
```diff
@@ -0,0 +1,345 @@
...
+}
+```
+
+## Trusted servers
+
+You can perform token exchange within a single custom authorization server or between other custom authorization servers under the same Okta tenant. The example discussed above is a token exchange within a single custom authorization server. When you want to perform token exchange between custom authorization servers within the same tenant, you need to make the authorization server that issued the subject token as trusted under the authorization server against which the token exchange request is made.
+
+A trusted server handles authenticated requests after the application obtains an access token. The incoming subject token (access token) is used to evaluate a subject.
+
+The following example walks you through how to set up a trusted server, scopes, access policy, and rule. Then, you can make the token exchange request with the trusted server.
```
Comment (isaacdurham-okta) [inline packages/@okta/vuepress-site/docs/guides/set-up-token-exchange/main/index.md:252]: Adjust this if you changed the earlier use of _walk you through_.

Comment (isaacdurham-okta) [inline packages/@okta/vuepress-site/docs/guides/set-up-token-exchange/main/index.md:252]:
```suggestion
The following example walks you through how to set up a trusted server, scopes, access policy, and rule. After you complete these steps, you can make the token exchange request with the trusted server.
```

File: packages/@okta/vuepress-site/docs/guides/set-up-token-exchange/main/index.md [hunk 14]
```diff
@@ -0,0 +1,345 @@
...
+
+### Add a trusted server
+
+Add the custom authorization server that you used in the previous flow as a trusted server of the authorization server that handles the token exchange in the trusted server flow.
+
+1. In the Admin Console, go to **Security** > **API**.
+1. Select the edit (pencil) icon on the right of the custom authorization server that you want to associate a trusted server with for this example.
+1. In the **Trusted servers** section, click **Add Server**.
+1. In the **Search** field, enter the name of the authorization server that you used in the token exchange flow previously. Matching results appear in a list. If more than 20 results appear, you can click **Show more**.
```
Comment (isaacdurham-okta) [inline packages/@okta/vuepress-site/docs/guides/set-up-token-exchange/main/index.md:261]:
```suggestion
1. In the **Search** field, enter the name of the authorization server that you used in the previous token exchange flow. Matching results appear in a list. If more than 20 results appear, you can click **Show more**.
```

File: packages/@okta/vuepress-site/docs/guides/set-up-token-exchange/main/index.md [hunk 15]
```diff
@@ -0,0 +1,345 @@
...
+1. Click **Add** beside the authorization server, and then click *Done**. The authorization server appears in the **Trusted servers** section.
```
Comment (isaacdurham-okta) [inline packages/@okta/vuepress-site/docs/guides/set-up-token-exchange/main/index.md:262]:
```suggestion
1. Click **Add** beside the authorization server, and then click **Done**. The authorization server appears in the **Trusted servers** section.
```

File: packages/@okta/vuepress-site/docs/guides/set-up-token-exchange/main/index.md [hunk 16]
```diff
@@ -0,0 +1,345 @@
...
+
+#### Create access policies and rules
+
+An [access policy](/docs/guides/configure-access-policy/main/) helps you secure your APIs by defining different access token lifetimes for a given combination of the grant type, user, and scope. You create policy rules to determine if an application should be permitted to access specific information from your protected APIs and for how long. Access policies are specific to a particular authorization server, the client applications that you designate for the policy, and the API resource for which you want to mint tokens.
+
+1. From the **API** page, select the edit (pencil) icon for the authorization server that you want to use for the initial authentication.
+1. Select the **Access Policies** tab, and then click **Add New Access Policy** to add a policy that allows the native app to access API1.
+1. In the **Add Policy** dialog that appears, enter the following:
+
+   * **Name:** Enter a name for the new access policy: **Access API1**.
```
Comment (isaacdurham-okta) [inline packages/@okta/vuepress-site/docs/guides/set-up-token-exchange/main/index.md:113]:
```suggestion
   * **Name:** Enter **Access API1**.
```

File: packages/@okta/vuepress-site/docs/guides/set-up-token-exchange/main/index.md [hunk 17]
```diff
@@ -0,0 +1,345 @@
...
+1. Click **Create Policy**.
+1. Click **Add Rule** and in the dialog that appears, enter the following:
+
+   * **Name:** Enter a name for the rule: **Mobile app to API1**.
+   * **AND Scopes requested:** Select **The following scopes** and enter **openid**.
+
+1. Click **Create rule**.
+1. Repeat steps 1-5 to create a policy and a rule that allows the service app that represents API1 to talk to API2. Use the following values for the policy:
+
+   * **Name:** **Access API2**.
```
Comment (isaacdurham-okta) [inline packages/@okta/vuepress-site/docs/guides/set-up-token-exchange/main/index.md:126]:
```suggestion
   * **Name:** Enter **Access API2**.
```

Comment (isaacdurham-okta) [reply packages/@okta/vuepress-site/docs/guides/set-up-token-exchange/main/index.md:126]: Also, this one doesn't need a description like step 3?

File: packages/@okta/vuepress-site/docs/guides/set-up-token-exchange/main/index.md [hunk 18]
```diff
@@ -0,0 +1,345 @@
...
+1. Click **Add** beside the authorization server, and then click *Done**. The authorization server appears in the **Trusted servers** section.
+
+### Create an access policy and rule
+
+Create an access policy and rule so that the service app can access API2 using a different authorization server than the one that granted the original access token for the user.
+
+1. Select the Access Policies tab, and then click Add New Access Policy to add a policy that allows the service app to access API2.
+1. In the Add Policy dialog that appears, enter the following:
+
+   * **Name:** Enter a name for the new access policy (**Access API2**).
```
Comment (isaacdurham-okta) [inline packages/@okta/vuepress-site/docs/guides/set-up-token-exchange/main/index.md:271]:
```suggestion
   * **Name:** Enter **Access API2**.
```

Comment (isaacdurham-okta) [reply packages/@okta/vuepress-site/docs/guides/set-up-token-exchange/main/index.md:271]: This format should match the earlier steps.

File: packages/@okta/vuepress-site/docs/guides/set-up-token-exchange/main/index.md [hunk 19]
```diff
@@ -0,0 +1,345 @@
...
+   * **Description:** Enter a description.
```
Comment (isaacdurham-okta) [inline packages/@okta/vuepress-site/docs/guides/set-up-token-exchange/main/index.md:272]: Match this to whatever format you decide for the **Create access policies and rules** section.

File: packages/@okta/vuepress-site/docs/guides/set-up-token-exchange/main/index.md [hunk 20]
```diff
@@ -0,0 +1,345 @@
...
+   * **Assign to:** Select **The following clients**, start typing **API1**, and then select it from the list that appears.
+
+1. Click **Create Policy**.
+1. Click **Add Rule** and in the dialog that appears, enter the following:
+
+   * Name: Enter a name for the rule (**API1 to API2**).
+   * AND Scopes requested: Select The following scopes, start typing **api:access:read** and select it from the list that appears. Repeat for **api:access:write** and select it from the list.
```
Comment (isaacdurham-okta) [inline packages/@okta/vuepress-site/docs/guides/set-up-token-exchange/main/index.md:279]:
```suggestion
   * **AND Scopes requested**: Select **The following scopes**, start typing **api:access:read** and select it from the list that appears. Repeat for **api:access:write** and select it from the list.
```

File: packages/@okta/vuepress-site/docs/guides/set-up-token-exchange/main/index.md [hunk 21]
```diff
@@ -0,0 +1,345 @@
...
+1. In the Add Policy dialog that appears, enter the following:
+
+   * **Name:** Enter a name for the new access policy (**Access API2**).
+   * **Description:** Enter a description.
+   * **Assign to:** Select **The following clients**, start typing **API1**, and then select it from the list that appears.
+
+1. Click **Create Policy**.
+1. Click **Add Rule** and in the dialog that appears, enter the following:
+
+   * Name: Enter a name for the rule (**API1 to API2**).
```
Comment (isaacdurham-okta) [inline packages/@okta/vuepress-site/docs/guides/set-up-token-exchange/main/index.md:278]: Match this to whatever format you decide for the Create access policies and rules section.

File: packages/@okta/vuepress-site/docs/guides/set-up-token-exchange/main/index.md [hunk 22]
```diff
@@ -0,0 +1,345 @@
...
+1. In the **Add Policy** dialog that appears, enter the following:
+
+   * **Name:** Enter a name for the new access policy: **Access API1**.
+   * **Description:** Enter a description.
+   * **Assign to:** Select **The following clients**, start typing the name of the native app that you created earlier, and select it from the list that appears.
+
+1. Click **Create Policy**.
+1. Click **Add Rule** and in the dialog that appears, enter the following:
+
+   * **Name:** Enter a name for the rule: **Mobile app to API1**.
```
Comment (isaacdurham-okta) [inline packages/@okta/vuepress-site/docs/guides/set-up-token-exchange/main/index.md:120]:
```suggestion
   * **Name:** Enter **Mobile app to API1**.
```

File: packages/@okta/vuepress-site/docs/guides/set-up-token-exchange/main/index.md [hunk 23]
```diff
@@ -0,0 +1,345 @@
...
+   * **AND Scopes requested:** Select **The following scopes** and enter **openid**.
+
+1. Click **Create rule**.
+1. Repeat steps 1-5 to create a policy and a rule that allows the service app that represents API1 to talk to API2. Use the following values for the policy:
+
+   * **Name:** **Access API2**.
+   * **Assign to:** Select **The following clients**, start typing **API1** (the service app that you created earlier) and select it from the list that appears.
+
+   Use the following values for the rule:
+   * **Name:** For this example, enter: **API1 to API2**.
```
Comment (isaacdurham-okta) [inline packages/@okta/vuepress-site/docs/guides/set-up-token-exchange/main/index.md:130]:
```suggestion
   * **Name:** Enter **API1 to API2**.
```

File: packages/@okta/vuepress-site/docs/guides/set-up-token-exchange/main/index.md [hunk 24]
```diff
@@ -0,0 +1,345 @@
...
+    "token_type": "Bearer",
+    "expires_in": 3600,
+    "access_token": "eyJraWQiOiJ6………X1Z4QA",
+    "scope": "openid"
+}
+```
+
+### Token exchange request from service app to api
+
+> **Note:** Use the Base64 encoded client ID and secret for the service app that you generated during the service app creation earlier. Place that in the Authorization header. See the following example for the format.
```
Comment (isaacdurham-okta) [inline packages/@okta/vuepress-site/docs/guides/set-up-token-exchange/main/index.md:186]:
```suggestion
> **Note:** You must include the Base64 encoded client ID and secret within the Authorization header. See the following example for the format.
```

File: packages/@okta/vuepress-site/docs/guides/set-up-token-exchange/main/index.md [hunk 25]
```diff
@@ -0,0 +1,345 @@
...
+    "expires_in": 3600,
+    "access_token": "eyJraWQiOiJR.....ajXZIk3GryDPC8OIhLsQ",
+    "scope": "api:access:read api:access:write",
+    "issued_token_type": "urn:ietf:params:oauth:token-type:access_token"
+}
+```
+
+**Access token decoded**
+
+The decoded access token shows the audience of the custom authorization server, the scopes requested, and the original user as the `sub`.
```
Comment (isaacdurham-okta) [inline packages/@okta/vuepress-site/docs/guides/set-up-token-exchange/main/index.md:326]:
```suggestion
The decoded access token shows the audience of the custom authorization server, the requested scopes, and the original user in the `sub` parameter.
```

File: packages/@okta/vuepress-site/docs/guides/set-up-token-exchange/main/index.md [hunk 26]
```diff
@@ -0,0 +1,345 @@
...
+   * Name: Enter a name for the rule (**API1 to API2**).
+   * AND Scopes requested: Select The following scopes, start typing **api:access:read** and select it from the list that appears. Repeat for **api:access:write** and select it from the list.
+
+1. Click **Create rule**.
+
+### Flow specifics with a trusted server
+
+Perform the requests in the previous [Flow specifics](#flow-specifics) section. When you reach the [token exchange request from the service app](#token-exchange-request-from-service-app-to-api) request, it should look like the following:
+
+> **Note:** Use the Base64 encoded client ID and secret for the service app that you generated during the service app creation earlier. Place that in the Authorization header. See the following example for the format.
```
Comment (isaacdurham-okta) [inline packages/@okta/vuepress-site/docs/guides/set-up-token-exchange/main/index.md:287]: Match any changes to the other note.

Comment (isaacdurham-okta) [reply packages/@okta/vuepress-site/docs/guides/set-up-token-exchange/main/index.md:287]: On a second look, is this note even needed? It looks like this codeblock is an example of what the user would see, not something they'd need to do themselves.

File: packages/@okta/vuepress-site/docs/guides/set-up-token-exchange/main/index.md [hunk 27]
```diff
@@ -0,0 +1,345 @@
...
+1. Select **API Services**, and then click **Next**.
+1. Name your application. For this example, enter **API1**, and then click **Save**.
+1. In the **General Settings** section of the **General** tab, click **Edit** and select **Token Exchange** as the grant type for your application.
+1. Click **Save**.
+1. Copy the client ID in the **Client Credentials** section, and then copy the client secret in the **CLIENT SECRETS** section.
+1. [Base64 encode](/docs/guides/implement-grant-type/clientcreds/main/#base64-encode-the-client-id-and-client-secret) the client ID and client secret for use in the token exchange request from API1 to API2.
+
+### Update the custom authorization servers
+
+You need to complete a few tasks to update the custom authorization servers that you plan to use for this example.
```
Comment (isaacdurham-okta) [inline packages/@okta/vuepress-site/docs/guides/set-up-token-exchange/main/index.md:81]:
```suggestion
The following tasks explain how to update the custom authorization servers.
```

Comment (isaacdurham-okta) [reply packages/@okta/vuepress-site/docs/guides/set-up-token-exchange/main/index.md:81]: Something other than _tasks_ might work too.

File: packages/@okta/vuepress-site/docs/guides/set-up-token-exchange/main/index.md [hunk 28]
```diff
@@ -0,0 +1,345 @@
...
+#### Create custom scopes
+
+Add some custom scopes to both custom authorization servers to request during the token exchange calls.
+
+> **Note:** You can't include the `offline_access` scope or any of the OpenID Connect scopes when using a service app. This means that you can't request refresh tokens or ID tokens in a service app-initiated token exchange flow.
+
+1. From the Admin Console, select **Security** > **API**, and then select one of the authorization servers that you want to use.
+1. Select the **Scopes** tab, and then click **Add Scope**.
+1. In the dialog that appears, enter a scope name (**api:access:read**).
+1. Click Create.
```
Comment (isaacdurham-okta) [inline packages/@okta/vuepress-site/docs/guides/set-up-token-exchange/main/index.md:92]:
```suggestion
1. Click **Create**.
```

File: packages/@okta/vuepress-site/docs/guides/set-up-token-exchange/main/index.md [hunk 29]
```diff
@@ -0,0 +1,345 @@
...
+1. Select the edit (pencil) icon on the right of the custom authorization server that you want to associate a trusted server with for this example.
+1. In the **Trusted servers** section, click **Add Server**.
+1. In the **Search** field, enter the name of the authorization server that you used in the token exchange flow previously. Matching results appear in a list. If more than 20 results appear, you can click **Show more**.
+1. Click **Add** beside the authorization server, and then click *Done**. The authorization server appears in the **Trusted servers** section.
+
+### Create an access policy and rule
+
+Create an access policy and rule so that the service app can access API2 using a different authorization server than the one that granted the original access token for the user.
+
+1. Select the Access Policies tab, and then click Add New Access Policy to add a policy that allows the service app to access API2.
```
Comment (isaacdurham-okta) [inline packages/@okta/vuepress-site/docs/guides/set-up-token-exchange/main/index.md:268]:
```suggestion
1. Select the Access Policies tab, and then click **Add New Access Policy** to add a policy that allows the service app to access API2.
```

File: packages/@okta/vuepress-site/docs/guides/set-up-token-exchange/main/index.md [hunk 30]
```diff
@@ -0,0 +1,345 @@
...
+
+> **Note:** This flow assumes that authentication and authorization of the user are complete and an access token and ID token have been issued by the Okta authorization server.
+
+1. The user is successfully signed in to a mobile app and makes a request.
+1. The mobile app includes the user’s valid access token in a request to the API1 service.
+1. API1 needs to talk to API2 to further process the user request. The API1 makes a request to the Okta authorization server to exchange the user’s access token for a new token with appropriate scopes to make requests to the API2 service on behalf of the user. This helps retain the user context.
+1. The Okta authorization server validates the user’s access token and grants a new access token to the API1 service with scopes that allow that service to make requests to the API2 service. The new access token retains the user context so API2 service knows on whose behalf the request is being made.
+1. The API1 service makes a request to the API2 service with the new access token.
+
+> **Note:** You can determine which API service made API calls on behalf of which user by analyzing the Okta System Log or the looking for token grant events.
```
[the remark this answers is not in the document]
Comment (venkatviswanathan-okta) [reply packages/@okta/vuepress-site/docs/guides/set-up-token-exchange/main/index.md:47]: by analyzing the Okta System Log's token grant events.

File: packages/@okta/vuepress-site/docs/guides/set-up-token-exchange/main/index.md [hunk 31]
```diff
@@ -0,0 +1,345 @@
...
+
+## Set up token exchange
+
+The following sections walk you through the set up for an example token exchange flow using a single custom authorization server. For an example token exchange flow using more than one custom authorization server within an Okta tenant, see the [Trusted servers](#trusted-servers) section.
+
+> **Note:** You can use a service, web, native, and SPA app to exchange a token with a service app.
```
[the remark this answers is not in the document]
Comment (venkatviswanathan-okta) [reply packages/@okta/vuepress-site/docs/guides/set-up-token-exchange/main/index.md:53]: This not is confusing. Option 1: Remove it

Option 2: Rephrase it "A service app can perform token exchange using tokens minted for a SPA, Native or Web App"

PS: I prefer removing it.

File: packages/@okta/vuepress-site/docs/guides/set-up-token-exchange/main/index.md (hunk 17 above)
Comment (susanharper-okta) [reply packages/@okta/vuepress-site/docs/guides/set-up-token-exchange/main/index.md:126]: a description isn't a required field when you create a Rule.

File: packages/@okta/vuepress-site/docs/guides/set-up-token-exchange/main/index.md (hunk 26 above)
Comment (susanharper-okta) [reply packages/@okta/vuepress-site/docs/guides/set-up-token-exchange/main/index.md:287]: the following example is a POST request in cURL format. They add Basic and then the base64 encoded client id and secret. So, I'm not sure why the note shouldn't be included.

File: packages/@okta/vuepress-site/docs/guides/set-up-token-exchange/main/index.md [hunk 32]
```diff
@@ -0,0 +1,345 @@
...
+> **Note:** You can't include the `offline_access` scope or any of the OpenID Connect scopes when using a service app. This means that you can't request refresh tokens or ID tokens in a service app-initiated token exchange flow.
+
+1. From the Admin Console, select **Security** > **API**, and then select one of the authorization servers that you want to use.
+1. Select the **Scopes** tab, and then click **Add Scope**.
+1. In the dialog that appears, enter a scope name (**api:access:read**).
+1. Click Create.
+1. Repeat steps 1-4 and create a second custom scope called **api:access:write**.
+1. Repeat steps 1-5 for the second authorization server that you want to use.
+
+#### Update audience
```
Comment (venkatviswanathan-okta) [inline packages/@okta/vuepress-site/docs/guides/set-up-token-exchange/main/index.md:96]: This is not needed if you use 2 custom AS (There is no default AS)

**Review by venkatviswanathan-okta — APPROVED** [review]
LGTM 🚢

**Review by isaacdurham-okta — APPROVED** [review]

**Pull request merged**